### PR TITLE
Circular dependencies 

### LINKED ISSUE #3: Mark cyclic dependencies red
I hate dependency cycles. I like to get rid of them. I would like to use rubrowser to find them.
So what about an option to mark cyclic dependencies?

Cheers, ellcs

### PR DESCRIPTION
**Pull request by steven-solomon**
In this branch I have done the following:
1) Added basic test coverage through the Data class using RSpec
2) Implemented the feature
3) Added example file in lib to test behavior.

I am happy to discuss switching to another testing framework if you prefer, I simply chose what I am most comfortable with. Also, I am hoping for a better way to test the app then pollute it with the test classes.
Fixes #3

### CODE REVIEW

**Comment by steven-solomon** [comment]
Is there a way to run these checks locally? I don't want to have a million tiny commits.

**Comment by blazeeboy** [comment]
@steven-solomon Yup, try rubocop locally I think houndci is just running that command

File: spec/parser/data_spec.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
+require 'rubrowser/data'
+
+describe Rubrowser::Data do
```
Comment (steven-solomon) [inline spec/parser/data_spec.rb]: Can we make an exception to this rule for spec files?

File: lib/rubrowser/data.rb [hunk 2]
```diff
@@ -1,30 +1,74 @@
...
+
+    def mark_circular_components(components)
+      @definitions.each do |definition|
+        if components.include?(definition.namespace.first.to_s)
+          definition.set_circular
+        end
+      end
+
+      @relations.each do |relation|
+        if components.include?(relation.namespace.to_s)
```
Comment (steven-solomon) [inline lib/rubrowser/data.rb]: In this case, I feel that using &&/|| for control flow would make the code unclear. How do you feel?

File: lib/rubrowser/a.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class A
+  def initialize
+    B.do
+  end
+
+  def self.do; end
+end
+
+class B
```
Comment (steven-solomon) [inline lib/rubrowser/a.rb]: I'd love to find another way to test the app's circular functionality without including a file that does nothing

**Comment by blazeeboy** [comment]
That's a good idea, lets add these exceptions to rubocop config file

File: spec/parser/data_spec.rb (hunk 1 above)
Comment (blazeeboy) [inline spec/parser/data_spec.rb]: sure

**Comment by steven-solomon** [comment]
Seems like houndci is ignoring the changes to .rubocop.yml. I'm going to close and reopen this pull request.

**Pull request closed without merging**